### PR TITLE
Gem Blog app: add devise

### PR DESCRIPTION
**Pull request by tomdan-ai**
## In This Milestone

**The Team Implemented the following:**
- Remove `current_user` method in `ApplicationController`, 
- [Install and setup devise](https://github.com/heartcombo/devise#getting-started).
  - Can register a new user.
  - User logs in with a combination of email and password.
  - Hashed password should be stored in the database.
  - Can reset password.
- [Modify the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A) and your styling.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,16 +1,17 @@
 # app/models/user.rb
 
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (Gambit142) [inline app/models/user.rb]: - [ ] One of the requirements for this project is to ensure that the application asks users who try to sign up to confirm their email. You have not enabled that module of `devise`. Kindly enable the `:confirmable` module and configure the confirmation of email correctly. 👍 😎

File: app/views/devise/registrations/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (Gambit142) [inline app/views/devise/registrations/new.html.erb]: - [ ] One of the requirements for this milestone is to design the `signup` page to look identical to that provided in the wireframe. Currently, the input boxes and the submit button is not identical to the view in the wireframe. Kindly redesign it properly. 👍 💻 

### Your Design

image

### Wireframe Template

image

File: app/views/devise/sessions/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Log in" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (Gambit142) [inline app/views/devise/sessions/new.html.erb]: - [ ] One of the requirements for this milestone is to design the `login` page to look identical to that provided in the wireframe. Currently, the input boxes and the submit button is not identical to the view in the wireframe. Kindly redesign it properly. 👍 💻 

### Your Design

image

### Wireframe Template

image

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @tomdan-ai @Victoria129 , 

Good job so far!

## Highlights
- All linters checks passed ✔️ 
- Installed `devise` gem successfully ✔️ 
- Configured the `:registerable` module ✔️
- Configured the `:database_authenticable` module ✔️
- Configured the `:recoverable` module ✔️

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by AshakaE — APPROVED** [review]
Hi @tomdan-ai   , 

### Highlights ✨ 

- Thanks for making the requested changes 👍 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Proceed to the next level 👏 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**